dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx

```diff
@@ -61,45 +83,111 @@ export const Gauge = ({ visualizations, layout, config }: any) => {
+            //     threshold: {
+            //       line: { color: thresholds[0]?.color || 'red', width: 4 },
+            //       thickness: 0.75,
+            //       value: thresholds[0]?.value || 0,
+            //     },
+            //   }),
+
+            //threshold labels
+            ...(showThresholdLabels && thresholds && thresholds.length
+              ? {
+                  axis: {
+                    ticktext: [gauge.value, ...thresholds.map((t: ThresholdUnitType) => t.name)],
+                    tickvals: [gauge.value, ...thresholds.map((t: ThresholdUnitType) => t.value)],
+                    ticklen: 5,
+                  },
+                }
+              : {}),
+            // multiple threshold markers!!!!
+            ...(showThresholdMarkers &&
+              thresholds &&
+              thresholds.length && {
+                steps: thresholds.map((threshold: ThresholdUnitType) => {
+                  const value = Number(threshold.value);
+                  return {
+                    range: [value, value + 0.25] /*width needs improvement*/,
+                    color: threshold.color || 'red',
+                    name: threshold.name || '',
+                    visible: true,
+                  };
+                }),
+              }),
           },
         };
       });
     }
     return calculatedGaugeData;
-  }, [series, value, data, fields, thresholds]);
+  }, [
+    dimensions,
+    metrics,
+    data,
+    fields,
+    thresholds,
+    showThresholdMarkers,
+    orientation,
+    showThresholdLabels,
+    titleSize,
+    valueSize,
+  ]);
 
   const mergedLayout = useMemo(() => {
     const isAtleastOneFullRow = Math.floor(gaugeData.length / PLOTLY_GAUGE_COLUMN_NUMBER) > 0;
     return {
       grid: {
-        rows: Math.floor(gaugeData.length / PLOTLY_GAUGE_COLUMN_NUMBER) + 1,
-        columns: isAtleastOneFullRow ? PLOTLY_GAUGE_COLUMN_NUMBER : gaugeData.length,
+        ...(orientation === 'auto' || orientation === 'h'
+          ? {
+              rows: Math.floor(gaugeData.length / PLOTLY_GAUGE_COLUMN_NUMBER) + 1,
+              columns: isAtleastOneFullRow ? PLOTLY_GAUGE_COLUMN_NUMBER : gaugeData.length,
+            }
+          : {
+              columns: Math.floor(gaugeData.length / PLOTLY_GAUGE_COLUMN_NUMBER) + 1,
+              rows: isAtleastOneFullRow ? PLOTLY_GAUGE_COLUMN_NUMBER : gaugeData.length,
+            }),
         pattern: 'independent',
       },
       ...layout,
       ...(layoutConfig.layout && layoutConfig.layout),
       title: dataConfig?.panelOptions?.title || layoutConfig.layout?.title || '',
     };
-  }, [layout, gaugeData.length, layoutConfig.layout, dataConfig?.panelOptions?.title]);
+  }, [
+    data,
+    layout,
+    gaugeData.length,
+    layoutConfig.layout,
+    dataConfig?.panelOptions?.title,
+    orientation,
+  ]);
 
   const mergedConfigs = {
     ...config,
     ...(layoutConfig.config && layoutConfig.config),
   };
-
+  
```

Comment:
remove extra space added and prettify file.

Reply:
done